streaming/aggregate/avg_test.go

```diff
@@ -0,0 +1,254 @@
+package aggregate
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/dgraph-io/badger/v2"
+
+	"github.com/cube2222/octosql"
+	"github.com/cube2222/octosql/streaming/storage"
+)
+
+func TestAvgInt(t *testing.T) {
+	ctx := context.Background()
+	db, err := badger.Open(badger.DefaultOptions("test"))
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer func() {
+		db.Close()
+		os.RemoveAll("test")
+	}()
+
+	prefix := []byte("avg")
+
+	badgerStorage := storage.NewBadgerStorage(db)
+	tx := badgerStorage.BeginTransaction().WithPrefix(prefix)
+
+	aggr := NewAverageAggregate()
+
+	// Empty storage
+	ExpectValue(t, ctx, aggr, tx, octosql.MakeInt(0))
+
+	// AddValue
+	AddValue(t, ctx, aggr, tx, octosql.MakeInt(1)) // Val: 1	Num: 1
+
+	ExpectValue(t, ctx, aggr, tx, octosql.MakeFloat(1))
+
+	AddValue(t, ctx, aggr, tx, octosql.MakeInt(3)) // Val: 4	Num: 2
+
+	ExpectValue(t, ctx, aggr, tx, octosql.MakeFloat(2))
+
+	AddValue(t, ctx, aggr, tx, octosql.MakeInt(8)) // Val: 12	Num: 3
+
+	ExpectValue(t, ctx, aggr, tx, octosql.MakeFloat(4))
+
+	AddValue(t, ctx, aggr, tx, octosql.MakeInt(2)) // Val: 14	Num: 4
+
+	ExpectValue(t, ctx, aggr, tx, octosql.MakeFloat(3.5))
+
+	// Wrong type passed
+	AddValueError(t, ctx, aggr, tx, octosql.MakeDuration(1234))
+
+	AddValueError(t, ctx, aggr, tx, octosql.MakeTime(time.Now()))
+
+	AddValueError(t, ctx, aggr, tx, octosql.MakeFloat(123.123)) // TODO - is this alright? You can't sum different types
```

Comment:
Yeah I think that's a good approach. If someone wants to sum mixed types they can just cast using the Float function